src/Database/Type.php

```diff
@@ -138,6 +138,7 @@ public static function buildAll()
     public static function set($name, Type $instance)
     {
         static::$_builtTypes[$name] = $instance;
+        static::$_types[$name] = get_class($instance);
```

Comment:
Wait, shouldn't this just be 
```
static::$_types[$name] =$instance;
```
? `Type::build` will understand this as it will only construct a new entity when the value is a class. And this allows one to pass instances to `Type::set` that actually contain configuration data and/or have constructors that do not only accept `$name`.

Reply:
> And this allows one to pass instances to Type::set that actually contain configuration data and/or have constructors that do not only accept $name.

Nothing preventing you to do so now either.

You suggestion will work for 3.x but not 4.x where `$_types` only contains mapping to class name strings. My fix resolves the issue for both versions.

Reply:
Ah wait you are right; it will only loose this information when you call `clear` :)